src/generator.cpp

```diff
@@ -752,6 +773,48 @@ GeneratorInternalVariablePtr Generator::GeneratorImpl::generatorVariable(const V
     return internalVariable;
 }
 
+bool Generator::GeneratorImpl::sameOrEquivalentVariable(const VariablePtr &variable1,
+                                                        const VariablePtr &variable2)
+{
+    // Return whether the given variables are the same or are equivalent (be it
+    // directly or indirectly).
+
+    return (variable1 == variable2) || variable1->hasEquivalentVariable(variable2, true);
+}
+
+GeneratorVariablePtr Generator::GeneratorImpl::variableFirstOccurrence(const VariablePtr &variable,
```

Comment:
The items I raised in this method have been reverted?

Reply:
?? I don't believe I have reverted anything, so not quite sure what you are talking about here.